ui/src/app/pages/appAnalysis/apps-list.tsx

```diff
@@ -45,13 +47,38 @@ export const AppsList = withCurrentWorkspace()((props) => {
   const { workspace } = props;
   const [selectedApp, setSelectedApp] = useState('');
   const [showSelectAppModal, setShowSelectAppModal] = useState(false);
+  const [showJupyterModal, setShowJupyterModal] = useState(false);
```

Comment:
what do you think about making this more generic?  Like `showModal(Jupyter)` or `showModal(null)`?  `onJupyterAppSelect` could become `onAppSelect` instead.